src/commands/update.rs

```diff
@@ -1,60 +1,802 @@
+                    format!("Provided argument is not a valid resource record: {arg}").into(),
+                );
+            }
+        }
+        Ok(records)
+    }
+
+    fn parse_ttl(arg: &str) -> Result<Ttl, Error> {
+        Ok(Ttl::from_secs(
+            if let Some(ttl) = arg.strip_suffix('s') {
+                ttl.parse()
+            } else if let Some(ttl) = arg.strip_suffix('m') {
+                ttl.parse::<u32>().map(|t| t / 60)
+            } else if let Some(ttl) = arg.strip_suffix('h') {
+                ttl.parse::<u32>().map(|t| t / 3600)
+            } else if let Some(ttl) = arg.strip_suffix('d') {
+                ttl.parse::<u32>().map(|t| t / 86400)
+            } else if let Some(ttl) = arg.strip_suffix('w') {
+                ttl.parse::<u32>().map(|t| t / 604800)
+            } else if let Some(ttl) = arg.strip_suffix('M') {
+                ttl.parse::<u32>().map(|t| t / 2629746) // 30.436875 days
+            } else if let Some(ttl) = arg.strip_suffix('y') {
+                ttl.parse::<u32>().map(|t| t / 31556952) // 365.2425 days
+            } else {
+                arg.parse()
+            }
+            .map_err(|err| Error::from(format!("Invalid TTL: {err}")))?,
+        ))
+    }
+
+    /// Create the packet of the update message to send to the name servers
+    fn create_update_message(
+        &self,
+        zone: &Name<Vec<u8>>,
+        prerequisites: UpdatePrerequisites,
+    ) -> Result<Vec<u8>, Error> {
+        // UPDATE message sections:
+        // Zone Section (= Question),
+        // Prerequisite Section (= Answer)
+        // Update Section (= Authority)
+        // Additional (= Additional)
+
+        let mut message = MessageBuilder::new_vec();
+
+        let header = message.header_mut();
+        header.set_opcode(Opcode::UPDATE);
+        header.set_qr(false);
+
+        let mut zone_section = message.question();
+        zone_section
+            .push(Question::new(zone, Rtype::SOA, Class::IN))
+            .unwrap();
+
+        let mut prereq_section = zone_section.answer();
+        Self::insert_prerequisites(prerequisites, &mut prereq_section)?;
+
+        let mut update_section = prereq_section.authority();
+
+        match self.action {
+            UpdateAction::Add { rtype, ref rdata } => {
+                if rdata.is_empty() {
+                    return Err("Provide at least one RDATA item to add".into());
+                }
+                for r in rdata {
+                    let rdata = Self::parse_rdata(rtype, r)?;
+                    update_section
+                        .push(Self::create_rr_addition(
+                            self.domain.clone(),
+                            self.class,
+                            self.ttl,
+                            rdata,
+                        ))
+                        .map_err(|e| -> Error {
+                            format!("Failed to add RR to UPDATE message: {e}").into()
+                        })?
```

Comment:
With this pattern it should be possible to do

```suggestion
                        .map_err(|e| {
                            format!("Failed to add RR to UPDATE message: {e}")
                        })?
```
But it's not super important.